giles-eco/src/main/java/edu/asu/diging/gilesecosystem/web/service/handlers/AbstractFileHandler.java

```diff
@@ -13,71 +13,70 @@
-    
-    protected byte[] getFileContentFromUrl(URL url) throws IOException {
-        URLConnection con = url.openConnection();
-        
-        InputStream input = con.getInputStream();
-
-        byte[] buffer = new byte[4096];
-        
-        ByteArrayOutputStream byteOutput = new ByteArrayOutputStream();
-        BufferedOutputStream output = new BufferedOutputStream(byteOutput);
-       
-        int n = -1;
-        while ((n = input.read(buffer)) != -1) {
-            output.write(buffer, 0, n);
-        }
-        input.close();
-        output.flush();
-        output.close();
-        
-        byteOutput.flush();
-        byte[] bytes = byteOutput.toByteArray();
-        byteOutput.close();
-        return bytes;
-    }
-    
-    @Override
-    public byte[] getFileContent(IFile file) {
-        try {
-            String downloadUrl = nepomukService.getFileDownloadPath(file);
-            if (downloadUrl == null) {
-                return null;
-            }
-            
-            return fileContentHelper.getFileContentFromUrl(new URL(downloadUrl));
-        } catch (IOException e) {
-            messageHandler.handleMessage("Could not download file.", e, MessageType.ERROR);
-            return null;
-        } 
-    }
-    
-    @Override
-    public String getFileUrl(IFile file) {
-        String gilesUrl = propertyManager.getProperty(Properties.GILES_URL).trim();
-        String pdfEndpoint = propertyManager.getProperty(Properties.GILES_FILE_ENDPOINT).trim();
-        String contentSuffix = propertyManager.getProperty(Properties.GILES_FILE_CONTENT_SUFFIX).trim();
-        
-        return gilesUrl + pdfEndpoint + file.getId() + contentSuffix;
-    }
+
+	@Autowired
+	protected IPropertiesManager propertyManager;
+
+	@Autowired
+	protected IFileContentHelper fileContentHelper;
+
+	@Autowired
+	protected INepomukUrlService nepomukService;
+
+	@Autowired
+	private ISystemMessageHandler messageHandler;
+
+	protected byte[] getFileContentFromUrl(URL url) throws IOException {
+		URLConnection con = url.openConnection();
+
+		InputStream input = con.getInputStream();
+
+		byte[] buffer = new byte[4096];
+
+		ByteArrayOutputStream byteOutput = new ByteArrayOutputStream();
+		BufferedOutputStream output = new BufferedOutputStream(byteOutput);
+
+		int n = -1;
+		while ((n = input.read(buffer)) != -1) {
+			output.write(buffer, 0, n);
+		}
+		input.close();
+		output.flush();
+		output.close();
+
+		byteOutput.flush();
+		byte[] bytes = byteOutput.toByteArray();
+		byteOutput.close();
+		return bytes;
+	}
+
+	@Override
+	public byte[] getFileContent(IFile file) throws NoNepomukFoundException {
+		String downloadUrl;
+		try {
+		  downloadUrl = nepomukService.getFileDownloadPath(file);
+		  return fileContentHelper.getFileContentFromUrl(new URL(downloadUrl));
+	
+		} catch (IOException e) {
+		  messageHandler.handleMessage("Could not download file.", e, MessageType.ERROR);
+		  return null;
+		}
```

Comment:
there are two scenarios where the file content couldn't be downloaded, either there is no nepomuk instance found or there is an ioexception while downloading. Since both result in the same (not file content), they need to be treated the same by this method, either throw an exception in both cases to let the caller know something is wrong, or return null.